internal/server/store.go

```diff
@@ -5,21 +5,28 @@ import (
 	"d7024e_group04/internal/kademlia/contact"
 	"d7024e_group04/internal/kademlia/kademliaid"
 	pb "d7024e_group04/proto"
+	"fmt"
 	"log"
 )
 
-// TODO: Input validation, Tests
+// TODO: Input validation
 func (s *Server) Store(ctx context.Context, content *pb.StoreRequest) (*pb.StoreResult, error) {
 	log.Print("received store rpc")
 
-	// TODO kademliaidfrombytes
-	senderID := (kademliaid.KademliaID)(content.RequestingNode.ID)
+	senderID := kademliaid.NewKademliaIDFromBytes([20]byte(content.RequestingNode.ID))
 	senderContact := contact.NewContact(senderID, content.RequestingNode.IPWithPort)
 	s.routingTable.AddContact(senderContact)
 
 	key := content.Key
 	value := content.Value
 
+	if len(key) == 0 || len(value) == 0 {
```

Comment:
anything more we need to validate?

Reply:
shouldn't the key be the same length as the address space? So env.IDLength?